File: CS_MapOverlay/CG_MapOverlayDll/Face.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CG_MapOverlayDll {
    public class Face {
        public HalfEdge outerComponent;
        public List<HalfEdge> innerComponents = new List<HalfEdge>();
        public List<Face> parents = new List<Face>();

        public HalfEdge getOuterComponent() {
            return outerComponent;
        }

        public void addParents(Face face) {
            if (!parents.Contains(face)) parents.Add(face);
        }
    }
}

File: compgeometry/src/ComputationalGeometry/MapOverlay/Face.java
package ComputationalGeometry.MapOverlay;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by IntelliJ IDEA.
 * User: alex
 * Date: 3/4/13
 * Time: 2:45 AM
 */
class Face {
    public HalfEdge outerComponent;
    public List<HalfEdge> innerComponents = new ArrayList<HalfEdge>();
    public List<Face> parents = new ArrayList<Face>();

    public HalfEdge getOuterComponent() {
        return outerComponent;
    }

    public void addParents(Face face) {
        if (!parents.contains(face)) parents.add(face);
    }
}